clarity/src/vm/analysis/type_checker/contexts.rs

```diff
@@ -136,3 +152,46 @@ impl<'a> TypingContext<'a> {
         }
     }
 }
+
+/// Subroutine of [TypeMap::concretize]
+fn concretize_deep(ty: TypeSignature) -> CheckResult<TypeSignature> {
```

Comment:
Do you think we should modify `TypeSignature::concretize` to do this? Seems like maybe it should already be doing this part for us, right?

Reply:
You are very right, I should have thought a bit more about it. Will change that.

Reply:
I was actually wondering if it should be inside of `concretize`, not adding a new method, `concretize_deep`. But any way, we haven't found any problems that require that change, so let's not do that for the time being.

Reply:
Ah yes, I misread you first message. I do prefer to have another method then, to not change the current behavior of something already well established.